src/diagnostic-channel-publishers/src/postgres.pub.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+import {EventEmitter} from "events";
+
+// copy the pg.Result type: https://node-postgres.com/api/result
+export interface IPostgresResult {
+    rowCount: number;
+    command: string;
+}
+
+export interface IPostgresData {
+    query: {
+        text?: string;
+        plan?: string;
+        preparable?: {
+            text: string;
+            args: any[];
+        }
+    };
+    database: {
+        host: string;
+        port: string;
+    };
+    result?: IPostgresResult;
+    duration: number;
+    error?: Error;
+}
+
+type PostgresCallback = (err: Error, res: IPostgresResult) => any;
+
+function postgres6PatchFunction(originalPg, originalPgPath) {
+    const originalClientQuery = originalPg.Client.prototype.query;
+
+    // wherever the callback is passed, find it, save it, and remove it from the call
+    // to the the original .query() function
+    originalPg.Client.prototype.query = function query(config, values, callback) {
+        const data: IPostgresData = {
+            query: {},
+            database: {
+                host: this.connectionParameters.host,
+                port: this.connectionParameters.port,
+            },
+            result: null,
+            error: null,
+            duration: 0,
+        };
+        const start = process.hrtime();
+        let queryResult;
+
+        function patchCallback(cb?: PostgresCallback): PostgresCallback {
+            return channel.bindToContext(function trackingCallback(err: Error, res: IPostgresResult): any {
```

Comment:
I'm wondering if you need some logic here to detect already wrapped callbacks, and then unwrap/re-wrap them if necessary.   e.g., this probably won't work exactly as you intended...
```javascript
const callback = () => {...};
config = { callback };
postgress.query(config);  // this will wrap up config.callback
postgress.query(config); // this will wrap up config.callback again!
```